web/src/hooks/redux.js

```diff
@@ -106,8 +127,10 @@ export function useCurrentZoneExchangeKeys() {
         Object.keys(zoneData.exchange).forEach((k) => exchangeKeys.add(k));
       }
     });
-    return Array.from(exchangeKeys).sort();
-  }, [isConsumption, zoneHistory, currentZoneData]);
+    const exchangeKeysToDisplay = getExchangesToDisplay(zoneId, isAggregated, Array.from(exchangeKeys).sort());
```

Comment:
We could just compute exchanges for each zone once and store it somewhere right? I.e instead of having it as part of the hook functionality?

Reply:
Yes but this way it updates when the zone data changes and keeps the list of current zone exchange logic in one file.

Exchange data could be missing for different time periods etc.